bpf/rtt_tracker.h

```diff
@@ -1,143 +1,161 @@
-    // check reversed flow
-    fill_flow_seq_id(seq_id, pkt, seq, true);
-
-    u64 *prev_ts = (u64 *)bpf_map_lookup_elem(&flow_sequences, seq_id);
-    if (prev_ts != NULL) {
-        u64 rtt = pkt->current_ts - *prev_ts;
-        /**
-         * FIXME: Because of SAMPLING the way it is done if we miss one of SYN/SYN+ACK/ACK
-         * then we can get RTT values which are the process response time rather than actual RTT.
-         * This check below clears them out but needs to be modified with a better solution or change
-         * the algorithm for calculating RTT so it doesn't interact with SAMPLING like this.
-         */
-        if (rtt < MIN_RTT) {
-            return;
+static inline void rtt_fill_in_tcp(struct sk_buff *skb, flow_id *id, u16 *flags) {
+    u16 skb_transport_header = BPF_CORE_READ(skb, transport_header);
+    u8 *skb_head = BPF_CORE_READ(skb, head);
+    struct tcphdr tcp;
+    u16 sport, dport;
+
+    __builtin_memset(&tcp, 0, sizeof(tcp));
+
+    bpf_probe_read(&tcp, sizeof(tcp), (struct tcphdr *)(skb_head + skb_transport_header));
+    sport = bpf_ntohs(tcp.source);
+    dport = bpf_ntohs(tcp.dest);
+    id->src_port = sport;
+    id->dst_port = dport;
+    set_flags(&tcp, flags);
+    id->transport_protocol = IPPROTO_TCP;
+}
+
+static inline int rtt_lookup_and_update_flow(flow_id *id, u16 flags, u64 rtt) {
+    flow_metrics *aggregate_flow = bpf_map_lookup_elem(&aggregated_flows, id);
+    if (aggregate_flow != NULL) {
+        aggregate_flow->end_mono_time_ts = bpf_ktime_get_ns();
+        aggregate_flow->flags |= flags;
+        if (aggregate_flow->flow_rtt < rtt) {
+            aggregate_flow->flow_rtt = rtt;
         }
-        pkt->rtt = rtt;
-        // Delete the flow from flow sequence map so if it
-        // restarts we have a new RTT calculation.
-        long ret = bpf_map_delete_elem(&flow_sequences, seq_id);
+        long ret = bpf_map_update_elem(&aggregated_flows, id, aggregate_flow, BPF_ANY);
         if (trace_messages && ret != 0) {
-            bpf_printk("error evicting flow sequence: %d", ret);
+            bpf_printk("error rtt updating flow %d\n", ret);
         }
-        // This is an ACK packet with valid sequence id so a SYN must
-        // have been sent. We can safely update the reverse flow RTT here.
-        update_reverse_flow_rtt(pkt, seq);
+        return 0;
     }
-    return;
-}
+    return -1;
+ }
 
-static __always_inline void __store_tcp_ts(pkt_info *pkt, struct tcphdr *tcp, flow_seq_id *seq_id) {
-    // store timestamp of syn packets.
-    u32 seq = bpf_ntohl(tcp->seq);
-    fill_flow_seq_id(seq_id, pkt, seq, false);
-    long ret = bpf_map_update_elem(&flow_sequences, seq_id, &pkt->current_ts, BPF_NOEXIST);
-    if (trace_messages && ret != 0) {
-        bpf_printk("err saving flow sequence record %d", ret);
-    }
-    return;
-}
+static inline int calculate_flow_rtt_tcp(struct sock *sk, struct sk_buff *skb) {
+	struct tcp_sock *ts;
```

Comment:
nit: don't we have a linter for C code format? looks like there's a double tab here. Not important.

Reply:
not really we don't have a C linter in the mix probably something nice to have I can look for one later, fixed this instance